zookeeper-server/src/main/java/org/apache/zookeeper/server/SyncRequestProcessor.java

```diff
@@ -57,7 +58,7 @@
         new LinkedBlockingQueue<Request>();
     private final RequestProcessor nextProcessor;
 
-    private Thread snapInProcess = null;
+    private final Semaphore snapThreadMutex = new Semaphore(1);
```

Comment:
So, just FYI.  I wanted to make this change because of two reasons:

1. It removes the special-case 'null' check (line 131) when the first iteration occurs.
2. The last 'snapInProcess' variable is never cleared out so there is always a dangling pointer to the last instance that ran... i.e., it never gets GC'ed until the next time a snap process runs and kicks the previous reference out.